web/packages/teleport/src/Discover/SelectResource/utils/pins.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { UserPreferences } from 'gen-proto-ts/teleport/userpreferences/v1/userpreferences_pb';
+
+import { DiscoverGuideId } from 'teleport/services/userPreferences/discoverPreference';
+
+import { SelectResourceSpec } from '../resources';
+
+export const defaultPins = [
+  DiscoverGuideId.ConnectMyComputer,
+  DiscoverGuideId.Kubernetes,
+  DiscoverGuideId.ApplicationWebHttpProxy,
+  // TODO(kimlisa): add linux server, once they are all consolidated into one
+];
+
+/**
+ * Only returns the defaults of resources that is available to the user.
+ */
+export function getDefaultPins(availableResources: SelectResourceSpec[]) {
+  return availableResources
+    .filter(r => defaultPins.includes(r.id))
+    .map(r => r.id);
+}
+
+/**
+ * Goes through resources and sets the "pinned" field
+ * for each resource. Pinned field is set to true if it's found
+ * in the users discover resource preferences.
+ *
+ * @param preferences - user preference
+ * @param resources - list of available resource guides
+ * @returns resource specs with the "pinned" field set.
+ */
+export function addPinnedField(
```

Comment:
I'm not sure if adding the pinned field to every resource is the best approach. Wouldn't it be simpler to determine if a resource is pinned at render time? Right now, we're duplicating data (since pins are already stored in preferences) and adding extra complexity by having to update it whenever pins or resources change.

I can imagine that instead we would have a function like `isPinned(resource, pins)`.

What do you think?